pkg/script/installer.go

```diff
@@ -9,170 +9,144 @@ import (
 
 	"github.com/openshift/source-to-image/pkg/api"
 	"github.com/openshift/source-to-image/pkg/docker"
-	"github.com/openshift/source-to-image/pkg/errors"
+	// "github.com/openshift/source-to-image/pkg/errors"
 	"github.com/openshift/source-to-image/pkg/util"
 )
 
-const defaultInstallDir = "upload/scripts"
+// InstallResult strucutre describes the result of install operation
+type InstallResult struct {
+	Downloaded bool
+	Installed  bool
+	Required   bool
+	Error      error
+}
 
 // Installer interface is responsible for installing scripts needed to run the build
 type Installer interface {
-	DownloadAndInstall(scripts []api.Script, workingDir string, required bool) (bool, error)
+	Install(required, optional []api.Script, dstDir string) map[api.Script]*InstallResult
 }
 
-// NewInstaller returns a new instance of the default Installer implementation
-func NewInstaller(image, scriptsURL, dstDir string, docker docker.Docker) Installer {
-	handler := &handler{
+// New returns a new instance of the default Installer implementation
+func New(image string, scriptsURL string, docker docker.Docker) Installer {
+	return &installer{
 		image:      image,
-		destDir:    dstDir,
 		scriptsURL: scriptsURL,
 		docker:     docker,
 		downloader: util.NewDownloader(),
 		fs:         util.NewFileSystem(),
+		result:     make(map[api.Script]*InstallResult),
 	}
-	return &installer{handler}
 }
 
 type installer struct {
-	handler scriptHandler
-}
-
-type handler struct {
-	docker     docker.Docker
 	image      string
 	scriptsURL string
-	destDir    string
+	docker     docker.Docker
 	downloader util.Downloader
 	fs         util.FileSystem
+	result     map[api.Script]*InstallResult
```

Comment:
Just my opinion... I would think it'd be better to keep the result as a return value and not make it part of the handler. It would make it clearer inside the Install function where the result comes from.